stan/math/prim/fun/csr_u_to_z.hpp

```diff
@@ -23,7 +23,6 @@ namespace math {
  * @throw std::out_of_range if i is out of range.
  */
 inline int csr_u_to_z(const std::vector<int>& u, int i) {
-  check_positive("csr_u_to_z", "u.size()", u.size());
```

Comment:
This requires `u.size() > 0`. I think check range is checking `u.size() > 1 + error_index`, and `error_index = 1` by default I think.

Reply:
Yes it does. However the nett cherck will fail if `u.size()==0` and its error type - `std::out_of_range` makes more sense that `std::domain_error`, which would be thrown by `check_positive`. And tests also expect `std::out_of_range`. I had to change this as previously `check_positive` was implemented in a way that would never throw for unsigned types.